bson/src/main/org/bson/json/UuidStringValidator.java

```diff
@@ -0,0 +1,72 @@
+/*
+ * Copyright 2008-present MongoDB, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.bson.json;
+
+import java.util.BitSet;
+
+final class UuidStringValidator {
+    private static final BitSet HEX_CHARS;
+
+    static {
+        HEX_CHARS = new BitSet(256);
+        HEX_CHARS.set('0', '9' + 1);
+        HEX_CHARS.set('A', 'F' + 1);
+        HEX_CHARS.set('a', 'f' + 1);
+    }
+
+    private static void validateFourHexChars(final String str, final int startPos) {
+        if (!(HEX_CHARS.get(str.charAt(startPos))
+                && HEX_CHARS.get(str.charAt(startPos + 1))
+                && HEX_CHARS.get(str.charAt(startPos + 2))
+                && HEX_CHARS.get(str.charAt(startPos + 3)))) {
+            throw new IllegalArgumentException(String.format("Expected four hexadecimal characters in UUID string \"%s\" starting at "
+                    + "position %d", str, startPos));
+        }
+    }
+
+    private static void validateDash(final String str, final int pos) {
+        if (str.charAt(pos) != '-') {
+            throw new IllegalArgumentException(String.format("Expected dash in UUID string \"%s\" at position %d", str, pos));
+        }
+    }
+
+    // UUID strings must be in the form 73ffd264-44b3-4c69-90e8-e7d1dfc035d4, but UUID.fromString fails to detect hyphens in the wrong
```

Comment:
This comment is simply a question, not an objection to the changes in the current PR.

Q:
While I understand that we do not publish API docs for this method, what are the reasons to write docs for internal code not in the javadoc format? The downsides of such a practice are:
- it forces reviewers to check whether or not the documented code is a part of public API;
- it introduces inconsistency in how docs are written;
- it prevents reading the doc without jumping to the source code.

Reply:
Let's discuss offline

Reply:
Saved for a retro meeting.